internal/commands/detect.go

```diff
@@ -0,0 +1,179 @@
+package commands
+
+import (
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/buildpacks/pack/internal/config"
+	"github.com/buildpacks/pack/internal/style"
+	"github.com/buildpacks/pack/pkg/client"
+	"github.com/buildpacks/pack/pkg/image"
+	"github.com/buildpacks/pack/pkg/logging"
+)
+
+type DetectFlags struct {
+	AppPath        string
+	Builder        string
+	Registry       string
+	DescriptorPath string
+	LifecycleImage string
+	Volumes        []string
+	PreBuildpacks  []string
+	PostBuildpacks []string
+	Policy         string
+	Network        string
+	Env            []string
+	EnvFiles       []string
+	Buildpacks     []string
+	Extensions     []string
+	Workspace      string
+	GID            int
+	UID            int
+	Publish        bool
+	DockerHost     string
+	RunImage       string
+}
+
+// Run Detect phase of lifecycle against a source code
+func Detect(logger logging.Logger, cfg config.Config, packClient PackClient) *cobra.Command {
+	var flags DetectFlags
+
+	cmd := &cobra.Command{
+		Use:     "detect",
+		Args:    cobra.ExactArgs(0),
+		Short:   "Run the detect phase of buildpacks against your source code",
+		Example: "pack detect --path apps/test-app --builder cnbs/sample-builder:bionic",
+		Long: "Pack Detect uses Cloud Native Buildpacks to run the detect phase of buildpack groups against the source code.\n" +
```

Comment:
where the output is shown wasn't touched on specifically in the RFC. We can either output to stdout or write it to disk. What do you think?

Reply:
That's a good question - I think since we'd want both the log output from each buildpack's ./bin/detect as well as the resulting group.toml and plan.toml, writing to disk makes the most sense. `pack build` has `--report-output-dir` and `--sbom-output-dir` configurable options, maybe we make a `--detect-output-dir`?

Reply:
Yes, I included the `--detect-output-dir` in the corresponding RFC. So that will be the way to go